src/sqlalchemy_cratedb/dialect.py

```diff
@@ -228,8 +228,12 @@ def connect(self, host=None, port=None, *args, **kwargs):
         servers = to_list(server)
         if servers:
             use_ssl = asbool(kwargs.pop("ssl", False))
-            if use_ssl:
+            # TODO: Switch to the canonical default `sslmode=prefer` later.
+            sslmode = kwargs.pop("sslmode", "disable")
+            if use_ssl or sslmode in ["allow", "prefer", "require", "verify-ca", "verify-full"]:
```

Comment:
If `ssl` is explicit disabled, `sslmode` shouldn't have any effect. At least for bwc imho.
```suggestion
            if use_ssl and sslmode in ["allow", "prefer", "require", "verify-ca", "verify-full"]:
```

Reply:
The patch is intended to set the stage to transition to PostgreSQL conventions, so `sslmode` is intended to be handled in parallel / as an alternative to the `ssl` option parameter, which we can phase out in the future.

Reply:
If that is the intention, then we should deprecate `ssl` in the docs/changes, and clearly state that `sslmode` has higher precedence and overrides `ssl`.

Reply:
Yes, documentation needs to be added. I will reflect that correspondingly, but do not intend to deprecate the `ssl` option just yet, maybe just give an outlook about it.

If you think it is the right choice to officially deprecate it, but still keep it just for bwc reasons, sure I can do it timely.

Reply:
imho, it wouldn't make sense to keep 2 different mechanisms in the future, or do you see some value in that?